### PR TITLE
Add kafka topic to accept MAP

### PR DESCRIPTION
**Pull request by dan-du-car**
# PR Details
## Description
Add kafka topic to accept MAP
Update decode data router to forward decoded map data to topic.OdeMapJsonand topic.OdeMapTxPojo
Fixing spat payload setSpat method name

## Related Issue
N/A

## Motivation and Context
V2XHub message support

## How Has This Been Tested?
Local integration testing

- Run jpo-ode

- Send below Json data to kafka topic: topic.OdeRawEncodedMAPJson

{"MapMessageContent":[{"metadata": {"utctimestamp": "2020-11-30T23:45:24.913657Z"},"payload":"00121E38041180000138044CD8EA0F3D3FC4A600000100000000030AE0C0615C18"}]}
image

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[ODE Contributing Guide](https://github.com/usdot-jpo-ode/jpo-ode/blob/bugfix/Pull_request_template/docs/contributing_guide.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: jpo-ode-core/src/main/java/us/dot/its/jpo/ode/model/OdeSpatPayload.java [hunk 1]
```diff
@@ -38,7 +38,7 @@ public J2735SPAT getSpat() {
 	        return (J2735SPAT) getData();
 	    }
 
-	    public void setBsm(J2735SPAT spat) {
+	    public void setSpat(J2735SPAT spat) {
```
Comment (TonyEnglish) [inline jpo-ode-core/src/main/java/us/dot/its/jpo/ode/model/OdeSpatPayload.java:41]: Good catch, but this seems like a bug fix rather part of the MAP based update

Comment (dan-du-car) [reply jpo-ode-core/src/main/java/us/dot/its/jpo/ode/model/OdeSpatPayload.java:41]: Yeah, I found this bug while I was implementing MAP. Update the description to include fixing the spat payload method naming

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/J2735RestrictionAppliesTo.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    emissionCompliant, 
+    equippedBicycle,
+    weightCompliant,
+    heightCompliant,
+    pedestrians,
+    slowMovingPersons,
+    wheelchairUsers,
+    visualDisabilities,
+    audioDisabilities, 
+    otherUnknownDisabilities,
```
Comment (TonyEnglish) [inline jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/J2735RestrictionAppliesTo.java:17]: i think in Java the last member of the enum should have a ";" or nothing (but not a ",")

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/J2735SegmentAttributeXY.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	   partialCurbIntrusion      , 
+	   taperToLeft               , 
+	   taperToRight              , 
+	   taperToCenterLine         , 
+	   headInParking             , 
+	   freeParking               , 
+	   timeRestrictionsOnParking ,                                
+	   costToPark                , 
+	   midBlockCurbPresent       , 
+	   unEvenPavementPresent     ,
```
Comment (TonyEnglish) [inline jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/J2735SegmentAttributeXY.java:40]: final in list should have ";" or nothing i think i java

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/J2735SpeedLimitType.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    maxSpeedInConstructionZone, 
+    vehicleMinSpeed,
+    vehicleMaxSpeed,            
+    vehicleNightMaxSpeed,
+    truckMinSpeed,
+    truckMaxSpeed,
+    truckNightMaxSpeed,
+    vehiclesWithTrailersMinSpeed,
+    vehiclesWithTrailersMaxSpeed,
+    vehiclesWithTrailersNightMaxSpeed,
```
Comment (TonyEnglish) [inline jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/J2735SpeedLimitType.java:16]: same comment as above

Comment (dan-du-car) [reply jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/J2735SpeedLimitType.java]: The commas are removed

Comment (dan-du-car) [reply jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/J2735SpeedLimitType.java]: Addressed b8eb2e0

**Review by drewjj — APPROVED** [review]
Tested the new MAP support on a local Linux machine. Followed the test steps provided in the pull request description and everything was decoded successfully and published to the Kafka topics.

Looks good!

**Review by snallamothu — APPROVED** [review]

**Pull request merged**